packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.component.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import getMonth from 'date-fns/get_month';
+import getYear from 'date-fns/get_year';
+
+import theme from './DateTimePicker.scss';
+import DateTimeView from '../views/DateTimeView';
+import MonthYearView from '../views/MonthYearView';
+
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		// eslint-disable-next-line
+		console.warn(`UNSTABLE WARNING: The 'DateTimePicker' and all the sub components aren't ready to be used in Apps. Code can (will) change outside the release process until it's ready.`);
+
+		const selectedDate = props.selection && props.selection.date;
+		const selectedTime = props.selection && props.selection.time;
+
+		const initialCalendarDate = selectedDate === undefined
+			? new Date()
+			: selectedDate;
+
+		this.state = {
+			isDateTimeView: true,
+			calendar: {
+				monthIndex: getMonth(initialCalendarDate),
+				year: getYear(initialCalendarDate),
+			},
+			selectedDate,
+			selectedTime,
+		};
+
+		this.setDateTimeView = this.setView.bind(this, true);
+		this.setMonthYearView = this.setView.bind(this, false);
+		this.onSelectCalendarMonth = this.onSelectCalendarMonth.bind(this);
+		this.onSelectCalendarYear = this.onSelectCalendarYear.bind(this);
+		this.onSelectCalendarMonthYear = this.onSelectCalendarMonthYear.bind(this);
+		this.onSelectDate = this.onSelectDate.bind(this);
+		this.onSelectTime = this.onSelectTime.bind(this);
+	}
+
+	componentWillReceiveProps(nextProps) {
+		const propsUpdated = this.props.selection !== nextProps.selection;
+
+		if (!propsUpdated) {
```

Comment:
Why 'It should' ? :D
I often declare variables to speak for the code/value it holds even if it's used only once.
I don't think of variable like it's there just for re-usability, but like it's hold the result of a computation even if it's intermediary.
In this way, it's not necessary to read the code for knowing what's done, just read the variables. I think it's ease the read with natural language.

Reply:
It is true that using proper constant names improve comprehension. 
But in this case you're making extra allocations that are not reused. And on top of it the constant is negated in the conditional invalidating the naming improvement.

Reply:
I have inverse the double negation for easy reading, I didn't notice it.
For the allocation and performance, yes it's a thing, but need to be relativized. We're not in a context with a billion record loop. It will be called only when user change the date via input or picker ; input is debounce and even if spam clicking the picker was possible (which is not, due to the submit feature), the impact will not be sensitive.